src/Artemis.UI/Screens/Debugger/Tabs/Logs/LogsDebugViewModel.cs

```diff
@@ -1,11 +1,74 @@
-﻿using Artemis.UI.Shared;
+﻿using Artemis.Core;
+using Artemis.UI.Shared;
+using Avalonia.Threading;
+using AvaloniaEdit.Document;
+using ReactiveUI;
+using Serilog.Events;
+using Serilog.Formatting.Display;
+using System.IO;
+using System.Reactive.Disposables;
 
 namespace Artemis.UI.Screens.Debugger.Logs;
 
 public class LogsDebugViewModel : ActivatableViewModelBase
 {
+    private readonly MessageTemplateTextFormatter _formatter;
+
+    public TextDocument Document { get; }
+
+    private const int MAX_ENTRIES = 1000;
+    
     public LogsDebugViewModel()
     {
         DisplayName = "Logs";
+        Document = new TextDocument();
+        _formatter = new MessageTemplateTextFormatter(
+            "[{Timestamp:yyyy-MM-dd HH:mm:ss.fff}] [{Level:u3}] [{SourceContext}] {Message:lj}{NewLine}{Exception}"
+        );
+
+        foreach(var logEvent in LogStore.Events)
+            AddLogEvent(logEvent);
+        
+        LogStore.EventAdded += OnLogEventAdded;
```

Comment:
Since you (correctly) unsubscribe when the VM deactivates it's best to subscribe when it activates and not in the constructor.
So simply move this down to line 36